### DISCUSSION TITLE
Why doesn't the bot get chat_join_request updates?

### DISCUSSION THREAD

**Discussion by yii-yu** [telegraf/telegraf#1688] [General]
Why doesn't the bot get chat_join_request updates when a new user joins the chat via an invite link? Is this a bug or am I missing something? The bot has all the permissions..

**Comment by MKRhere** [comment]
To receive `"chat_join_request"`, your invite link must be created using the "Create a new link" option in group settings with the "Request admin approval" option selected.

![image](https://user-images.githubusercontent.com/19621208/189976609-9bdeae75-c7d7-4c52-aa1f-e1d18e132d90.png)

If you just want to receive chat join updates, you must use the `"chat_member"` update instead. Refer https://core.telegram.org/bots/api#getting-updates.

**Reply by yii-yu** [reply]
> To receive `"chat_join_request"`, your invite link must be created using the "Create a new link" option in group settings with the "Request admin approval" option selected.
> 
> ![image](https://user-images.githubusercontent.com/19621208/189976609-9bdeae75-c7d7-4c52-aa1f-e1d18e132d90.png)
> 
> If you just want to receive chat join updates, you must use the `"chat_member"` update instead. Refer https://core.telegram.org/bots/api#getting-updates.

Thank you! You're right. Specified the creates_join_request:true parameter when creating an invite link using the createChatInviteLink method.

**Reply by MKRhere** [reply]
Yes! That's another way :)

**Reply by yii-yu** [reply]
In continuation of this topic, there is one more question. Joining public chats/channels via invitation link does not receive Invite_link to chatmemberupdated https://core.telegram.org/bots/api#chatmemberupdated. In private it comes in public it doesn't. Is that how it should be?
![image](https://user-images.githubusercontent.com/16443720/190852587-ca2c47a9-07a5-4ec6-bb5c-e8fe3f9f5cb2.png)

**Reply by MKRhere** [reply]
That is perhaps because invite links to public groups just opens the group, and the user has the option to join the group as if they never had an invite link in the first place?

**Reply by yii-yu** [reply]
That's the way it is. But it is quite logical to optionally send an Invite_link to chat_member if the user joined the public chat via a link. Maybe there is a way to find out which link the user has joined the public group?
